omnibus/files/private-chef-cookbooks/private-chef/libraries/private_chef.rb

```diff
@@ -810,9 +814,7 @@ def generate_config(node_name)
 
       generate_config_for_topology(PrivateChef["topology"], node_name)
 
-      unless PrivateChef["ldap"].nil? || PrivateChef["ldap"].empty?
-        gen_ldap
-      end
+      gen_ldap if PrivateChef["ldap"]["enabled"]
```

Comment:
The ldap_authentication_enabled? function is used at a number of call sites:

```
sdanna@thrace ~/oc/code/opscode/chef-server (master %) > rg 'ldap_authentication_enabled'
omnibus/files/private-chef-ctl-commands/password.rb
36:    if password == '' && ldap_authentication_enabled?
43:    message = if ldap_authentication_enabled?
53:    if ldap_authentication_enabled?
72:def ldap_authentication_enabled?

omnibus/files/private-chef-cookbooks/private-chef/libraries/helper.rb
192:  def ldap_authentication_enabled?

omnibus/files/private-chef-cookbooks/private-chef/recipes/opscode-erchef.rb
27:ldap_authentication_enabled = OmnibusHelper.new(node).ldap_authentication_enabled?
29:enable_ssl = ldap_authentication_enabled ? node['private_chef']['ldap']['enable_ssl'] : nil
30:ldap_encryption_type = ldap_authentication_enabled ? node['private_chef']['ldap']['encryption_type'] : nil
47:  variables(node['private_chef']['opscode-erchef'].to_hash.merge(:ldap_enabled => ldap_authentication_enabled,
```

We don't use it here because at this point in the process, the PrivateChef configuration object hasn't been merged into the node object.